eng/common/scripts/Verify-RestApiSpecLocation.ps1

```diff
@@ -0,0 +1,216 @@
+    return
+  }
+  try {
+    $searchResult = Search-GitHubCommit -AuthToken $GitHubPat -CommitHash $commit -RepoOwner "Azure" -RepoName "azure-rest-api-specs"
+    if ($searchResult.total_count -lt 1) {
+      LogError "Commit $commit doesn't exist in 'main' branch of Azure/azure-rest-api-specs repository. ServiceDir:$ServiceDirectory, PackageName:$PackageName"
+      exit 1
+    }
+    else{
+      Write-Host "ServiceDir:$ServiceDirectory, PackageName:$PackageName. Commit $commit exists in 'main' branch of Azure/azure-rest-api-specs repository."
+    }
+  }
+  catch {
+    LogError "ServiceDir:$ServiceDirectory, PackageName:$PackageName. Failed to search commit $commit with exception:`n$_ "
+    exit 1
+  }
+}
+
+function Verify-YamlContent([string]$markdownContent) {
+  $splitString = '``` yaml|```yaml|```'
+  $yamlContent = $markdownContent -split $splitString
+  foreach($yamlSection in $yamlContent) {
+    if ($yamlSection) {
+      try {
+        # remove the lines like: $(tag) == 'package-preview-2023-09'
+        $yamlSection = $yamlSection -replace '^\s*\$\(.+\)\s*==.+', ''
+        $yamlobj = ConvertFrom-Yaml -Yaml $yamlSection
+        if($yamlobj) {
+          $batchValue = $yamlobj["batch"]
+          $requireValue = $yamlobj["require"]
+          $inputFileValue = $yamlobj["input-file"]
+          if ($requireValue) {
+            LogDebug "require is set as:$requireValue"
+            Verify-Url $requireValue
+          }
+          elseif ($inputFileValue) {
+            LogDebug "input-file is set as:$inputFileValue"
+            foreach($inputFile in $inputFileValue) {
+              Verify-Url $inputFile
+            }
+          }
+          elseif ($batchValue) {
+            # there are some services which use batch mode for sdk generation, e.g. Azure.AI.Language.QuestionAnswering
+            foreach($batch in $batchValue) {
+              $requireValue = $batch["require"]
+              $inputFileValue = $batch["input-file"]
+              if ($requireValue) {
+                LogDebug "require is set as:$requireValue"
+                Verify-Url $requireValue
+              }
+              elseif ($inputFileValue) {
+                LogDebug "input-file is set as:$inputFileValue"
+                foreach($inputFile in $inputFileValue) {
+                  Verify-Url $inputFile
+                }
+              }
+            }
+          }
+        }
+      }
+      catch {
+        Write-Host "ServiceDir:$ServiceDirectory, PackageName:$PackageName. Failed to parse yaml section $yamlSection with exception:`n$_ "
+      }
+    }
+  }
+}
+
+function Verify-PackageVersion() {
+  try {
+    $resolvedPath = Resolve-Path -Path $ArtifactLocation
+    if(-not (Test-Path -Path $resolvedPath -IsValid)) {
+      LogError "ServiceDir:$ServiceDirectory, PackageName:$PackageName. Invalid artifact location: $ArtifactLocation"
+      exit(1)
+    }
+    $pkgs, $parsePkgInfoFn = RetrievePackages -artifactLocation $resolvedPath
```

Comment:
Let's avoid using RetrievePackages here instead do something similar to what APIView does. See https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/Create-APIReview.ps1#L107C63-L107C63 it reads the metadata from the packageinfo.json file.

Reply:
I updated it and validated locally. Can you review? @weshaggard